S2 ard s3 sync

kieranricardo opened the pull request:
This PR adds the `nci_s2_upload_s3_v2`, `k8s_index_s2_nbart_odc`, and `k8s_index_s2_nbart_sandbox` with the new aws users, queues, and sns defined in the current Datakube PRs. Once those PRs are merged an AWS airflow conneciton needs to be created for the S2 upload user  with the connection id `dea_public_data_sentinel2_ard`

alexgleith commented:
Linting is failing :-)

alexgleith commented on dags/sentinel_2_nbart/k8s_index_s2_nbart_odc.py:
```diff
@@ -0,0 +1,96 @@
+"""
+# Sentinel-3 indexing automation
```
Sentinel-2

alexgleith commented on dags/sentinel_2_nbart/upload_s2_nbart.py:
```diff
@@ -0,0 +1,365 @@
+)
+
+import rasterio
+from eodatasets3 import serialise, validate, images, documents
```
Can we run some kind of automatic sorting of imports please. There's some duplication and confusing ordering.

alexgleith commented on dags/sentinel_2_nbart/upload_s2_nbart.py:
```diff
@@ -0,0 +1,365 @@
+WORK_DIR = Path('/g/data/v10/work/s2_nbart_rolling_archive')
+SNS_ARN = "arn:aws:sns:ap-southeast-2:060378307146:dea-public-data-sentinel-2-ard"
+
+S3 = None
```
?

alexgleith commented on dags/sentinel_2_nbart/upload_s2_nbart.py:
```diff
@@ -0,0 +1,365 @@
+    """
+    session = boto3.session.Session()
+
+    if not check_granule_uploaded(granule_id, session):
```
I would love if we were consistent with dataset/granule/scene. Let's pick one and just use that! My preference is `dataset`.

kieranricardo replied:
the `c3_to_s3_rolling..` uses the granule terminology, so I'll use granule here as well to make it consistent when I reuse some of the `c3_to_s3_rolling..` code

alexgleith replied:
I actually prefer scene. My heirarchy of naming preferences: scene > dataset > granule...

But even more important is consistency!

alexgleith commented on dags/sentinel_2_nbart/upload_s2_nbart.py:
```diff
@@ -0,0 +1,365 @@
+    """
+    setup_logging()
+
+    global S3
+    S3 = s3_client()
```
Don't like the global client...

alexgleith commented on dags/sentinel_2_nbart/upload_s2_nbart.py:
```diff
@@ -0,0 +1,365 @@
+    )
+    stac = json.dumps(stac, default=json_fallback, indent=4)
+
+    s3_dump(yaml.safe_dump(serialise.to_doc(eo3), default_flow_style=False), s3_path_eo3, S3)
```
This doesn't need an S3 object, it'll make its own if we leave it out.

alexgleith commented on dags/sentinel_2_nbart/upload_s2_nbart.py:
```diff
@@ -0,0 +1,365 @@
+    resource = session.client('sns', region_name='ap-southeast-2')
+    resource.publish(
+        TopicArn=SNS_ARN,
+        Message=json.dumps({'default': stac}),
```
Why is this in the `default` key?

alexgleith commented on dags/sentinel_2_nbart/upload_s2_nbart.py:
```diff
@@ -0,0 +1,365 @@
+    return stac, s3_path_stac
+
+
+def send_stac_sns(stac, session):
```
If we're pushing out STAC, we should be attaching attributes to the notification too.

Check this: https://github.com/GeoscienceAustralia/dea-airflow/blob/master/scripts/c3_to_s3_rolling.py#L155

alexgleith reviewed (changes requested):
A bunch of comments.

Can we please look at the C3 to S3 rolling script and share functionality as much as we can.

https://github.com/GeoscienceAustralia/dea-airflow/blob/master/scripts/c3_to_s3_rolling.py

kieranricardo commented:
hey @alexgleith, I think I responded to all of your review comments. Re. sharing code, I changed some of the `c3_rolling...` functions to be more general and import them into my code. I also added an extra task in the dag to upload the `c3_rolling..` script so that its functions can be used on the NCI.

fyi this seems to be breaking CI bc `upload_s2_nbart.py` is not in the same directory as `c3_rolling` (but they are in the same directory when they're uploaded on NCI so the code works). One fix would be to move the `upload_s2_nbart.py` file into scripts, but I personally like all the related files and DAGs being in the same folder. Whadya reckon?

kieranricardo commented:
@alexgleith I've got CI to pass now :)

alexgleith commented on scripts/upload_s2_nbart.py:
```diff
@@ -0,0 +1,318 @@
+    return eo3
+
+
+def note_measurement(
```
This is copied from EODatasets, right? Can you remind me why?

kieranricardo replied:
This optimises `note_measurement` to not read in all the tiff files if they're not being used to create the valid data polygon. In our case this speeds things up 20x! (bc file reads on the NCI are so slow)

I also put this into a eodatasets  PR (https://github.com/GeoscienceAustralia/eo-datasets/pull/129) but AFAIK they haven't done another release

alexgleith replied:
Ok, cool.

Let's hassle Jeremy for a release!

kieranricardo replied:
@alexgleith `deamodule` built and is now using the latest release of `eodatasets3`! I changed the code here to used the `eodatasets3` `.note_measurement` method and deleted mine :)

alexgleith reviewed (changes requested):
Is that function from EO Dataset definitely needed in here?
